### ISSUE TITLE
Include more headers with the iOS framework

### ISSUE THREAD

**Issue by lukaskollmer** [janeasystems/nodejs-mobile#17]
IMO the iOS framework should also include the following headers commonly used for node/v8 development:
- [`node_api.h`](https://github.com/nodejs/node-chakracore/blob/master/src/node_api.h)
- [`node_api_types.h`](https://github.com/nodejs/node-chakracore/blob/master/src/node_api_types.h)
- [`uv.h`](https://github.com/nodejs/node-chakracore/blob/master/deps/uv/include/uv.h)
- [`v8.h`](https://github.com/nodejs/node-chakracore/blob/master/deps/v8/include/v8.h)
- ...

As of right now, I have to manually fetch these headers from the repo and include them with my Xcode project

**Comment by jaimecbernardo** [comment]
Thank you for the issue, @lukaskollmer . This is definitely an improvement to take into account.
Our current take on this is that we should include a header files package with the releases or in a folder inside each platform's release package. Would this be helpful for you?

**Comment by lukaskollmer** [comment]
Yes that approach would definitely be helpful, but tbh just including them in the iOS framework would be a lot more useful, especially since you could always be sure that you're using the correct headers for the version of `libnode` you're building against.

**Comment by IMcD23** [comment]
+1 on adding it to the framework, as that is one of the helpful things that the framework bundle is built to do. Frameworks are essentially a combination of dylibs & bundles, since they contain both the dynamic library & resources such as headers, assets, and strings. The headers get stripped out automatically by Xcode before release.

**Comment by lukaskollmer** [comment]
Downloaded the new release ([v0.1.3](https://github.com/janeasystems/nodejs-mobile/releases/tag/nodejs-mobile-v0.1.3)) and added the included headers to my Xcode project.

There's a build error because [`chakra_ttd.h`](https://github.com/janeasystems/nodejs-mobile/blob/mobile-master/src/chakra_ttd.h) is missing (included from [`node.h:99`](https://github.com/janeasystems/nodejs-mobile/blob/mobile-master/src/node.h#L99)). You'll need to ship that as well with the framework

**Comment by IMcD23** [comment]
I played around with adding headers to the framework yesterday. What’s the minimal set that you would need? Would it be better to include all headers, or pick and choose?

**Comment by lukaskollmer** [comment]
I think including all is probably the best choice since you can't really know which ones people will want to use, but I'd say the minimal set that should be included all all headers needed to write native modules (`node_api.h`, `uv.h`, `v8.h`, etc)

Also, there are a lot that probably should be omitted for iOS targets since they seem to be platform specific

**Comment by jaimecbernardo** [comment]
@lukaskollmer Thanks for noticing the missing `chakra_ttd.h`. I've updated the iOS v0.1.3 release package to include that header.
Currently, the headers provided for iOS are the ones distributed for use by native modules in the `node-chakracore` v8.6.0 release, on which the iOS part of `nodejs-mobile` is based.

Our current use of the header files in the React Native plugin uses the `N-API` and `libuv` headers only, so it didn't affect our current use and was released unnoticed. Could you please let me know if this inclusion helped you on your use case?

Regarding the inclusion of the headers in the .framework, there's a recent PR by @IMcD23 ( https://github.com/janeasystems/nodejs-mobile/pull/18 ) where a good point is made that the current headers would be incompatible with `Swift`, since they have a C++ interface.

This means that minimal header set identification and a C interface wrapping for them would be needed if Swift compatibility is intended, and then when trying to integrate native modules that depend on any C++ interface function, my guess is that those wouldn't be able to use the headers inside the framework.

Until the project matures a bit more and we can understand all the use cases, I think keeping the headers out of the framework is a more flexible way to accommodate a greater number of distinct use cases. What are your thoughts on this?

I also think the headers should be part of the framework.
